### ISSUE TITLE
ncurses mouse api: crashes if not initialized

### ISSUE THREAD

**Issue by GitMensch** [wmcbrine/PDCurses#67]
To reproduce just place a
```c
	int mouse_interval = mouseinterval (-1);
```

somewhere before the PDCurses initialization. Result: SIGSEGV.
The [man page](https://linux.die.net/man/3/mouseinterval) says for this function similar for others):

> returns the previous interval value, unless the terminal was not initialized. In that case, it returns the maximum interval value (166).

I'm confident that the SIGSEGV roots in direct unchecked access to `SP->`, so the "correct thing to match the documentation" is to check it here (and possibly in other places) and return the maximum.
Note: the maximum in PDCurses is 1000 (seems arbitrary, in any case likely a good candidate for a define, maybe useful to switch to ncurses' value [if I'm not wrong it will use the maximum if you specify something higher, PDCurses ignores the value in this case]).

**Comment by wmcbrine** [comment]
Personally, I'd be OK with it crashing, since that tells you right away "oops, forgot initscr()". But, fine, I realize curses doesn't work that way in general. I've put in SP checks throughout mouse.c (to be committed later).

For this particular function, though, I'm not really seeing the logic of returning the maximum interval on no SP. I realize that's what ncurses does, but not why. Zero would seem to make at least as much sense, reflecting the fact that you're not going to get any click events (or any other input events).

**Comment by GitMensch** [comment]
> I'm not really seeing the logic of returning the maximum interval on no SP

Maybe that's a "proper way" to know the maximum up-front? I don't know, it was just something I've stumbled over.

**Comment by wmcbrine** [comment]
For now, I've decided to return ERR in this situation. https://github.com/wmcbrine/PDCurses/commit/7e04826f991db300a875768f093539d601669304

**Comment by Bill-Gray** [comment]
@wmcbrine 'fixed' this by having a slew of functions return ERR if called before `initscr()`.  However,  I agree strongly that (in most cases) the program should crash,  and certainly should do so when compiled for debug mode.  Calling,  for example,  `init_color()` in ncurses without `init_scr()` does not crash,  but it also has no effect.  If you do it,  you've done something wrong,  and could spend some time trying to figure it out if your mistake is simply ignored.
[For my fork, I've added an `assert(SP)` for each such case.](https://github.com/Bill-Gray/PDCurses/commit/466c0780f6fd00a02cfaa37e40be779988ed466e)  This -- at least in debug mode -- makes considerably more sense to me.  I also [revised `mouseinterval()` to follow ncurses' lead and return the maximum mouse interval](https://github.com/Bill-Gray/PDCurses/commit/3f9fcb8c71e916a698f00b34c007fe5c622f47ed) (which I made a defined constant and bumped up to 32767 milliseconds).
